frontend/src/metabase/query_builder/components/view/ViewHeader.jsx

```diff
@@ -93,12 +93,17 @@ export function ViewTitleHeader(props) {
   const previousQuery = usePrevious(query);
 
   useEffect(() => {
-    if (!question.isStructured() || !previousQuestion?.isStructured()) {
+    const { isNative } = Lib.queryDisplayInfo(query);
+    const isPreviousQuestionNative =
+      previousQuery && Lib.queryDisplayInfo(previousQuery).isNative;
+
+    if (isNative || isPreviousQuestionNative) {
       return;
     }
 
     const filtersCount = Lib.filters(query, -1).length;
-    const previousFiltersCount = Lib.filters(previousQuery, -1).length;
+    const previousFiltersCount =
+      previousQuery && Lib.filters(previousQuery, -1).length;
```

Comment:
It's good when "count" is always a number:

```suggestion
    const previousFiltersCount =
      previousQuery ? Lib.filters(previousQuery, -1).length : 0;
```

Reply:
I tried that.
Related E2E tests started exploding, and I could confirm in the UI that things don't work as intended.

At that point, I couldn't even dig into WHY. This particular solution does work.

Reply:
Thanks for checking!

> At that point, I couldn't even dig into WHY

Probably because `1 > undefined` is `false`.